packages/typescript-estree/src/convert.ts

```diff
@@ -2870,14 +2870,14 @@ export class Converter {
               constraint: this.convertChild(node.typeParameter.constraint),
               key: this.convertChild(node.typeParameter.name),
               nameType: this.convertChild(node.nameType) ?? null,
-              optional:
-                node.questionToken &&
-                (node.questionToken.kind === SyntaxKind.QuestionToken ||
-                  getTextForTokenKind(node.questionToken.kind)),
-              readonly:
-                node.readonlyToken &&
-                (node.readonlyToken.kind === SyntaxKind.ReadonlyKeyword ||
-                  getTextForTokenKind(node.readonlyToken.kind)),
+              optional: node.questionToken
+                ? node.questionToken.kind === SyntaxKind.QuestionToken ||
+                  getTextForTokenKind(node.questionToken.kind)
+                : false,
+              readonly: node.readonlyToken
+                ? node.readonlyToken.kind === SyntaxKind.ReadonlyKeyword ||
+                  getTextForTokenKind(node.readonlyToken.kind)
+                : undefined,
```

Comment:
> Note that `readonly` is an enum of `true`, `+` and `-`. So previously I proposed `undefined` as the default value. I guess `false` does no harm here.

Agreed that either could work, and also agreed on going with `undefined`. That feels to me like a more normal default when there are multiple truthy options (`'+'`, `'-'`). I can see why `false` might make sense as a counterpoint to `true`, but visually I think it makes less sense.